src/wayland-eglsurface.c

```diff
@@ -183,6 +269,15 @@ wlEglSendDamageEvent(WlEglSurface *surface, struct wl_event_queue *queue)
                           surface->ctx.currentBuffer,
                           surface->dx,
                           surface->dy);
+
+        /*
+         * Send our explicit sync acquire and release points. This needs to be done
+         * as part of the surface attach as it is a protocol error to specify these
+         * points without attaching a buffer in the same commit.
+         */
+        if (!send_explicit_sync_points(surface->wlEglDpy, surface, image)) {
+            return EGL_FALSE;
```

Comment:
This is a bug. There is no turning back after wl_surface_attach has been called. You must prepare the entire state before modifying any of it.